src/deploy.ts

```diff
@@ -9,17 +9,21 @@ import { upload } from './upload.ts'
 const CONTRACT_TEXT_PREFIX = 't|'
 const CONTRACT_MANIFEST_PREFIX = 'm|'
 
-export async function deploy (args: string[]) {
+export async function deploy (args: string[]): Promise<void> {
   const [urlOrDirOrSqliteFile, ...manifests] = args
   if (manifests.length === 0) throw new Error('missing url or manifests!')
   const toUpload = []
   for (const manifestPath of manifests) {
-    const json = JSON.parse(Deno.readTextFileSync(manifestPath))
-    const body = JSON.parse(json.body)
+    interface ContractBody {
+      contract: { file: string }
+      contractSlim?: { file: string }
+    }
+    const json = JSON.parse(Deno.readTextFileSync(manifestPath)) as { body: string }
+    const body = JSON.parse(json.body) as ContractBody
     const dirname = path.dirname(manifestPath)
-    toUpload.push(CONTRACT_TEXT_PREFIX + path.join(dirname, body.contract.file))
-    if (body.contractSlim) {
-      toUpload.push(CONTRACT_TEXT_PREFIX + path.join(dirname, body.contractSlim.file))
+    toUpload.push(CONTRACT_TEXT_PREFIX + String(path.join(dirname, body.contract.file)))
+    if (body.contractSlim !== undefined && body.contractSlim !== null) {
+      toUpload.push(CONTRACT_TEXT_PREFIX + String(path.join(dirname, body.contractSlim.file)))
```

Comment:
Could these changes be reverted?

- The `!== undefined` and `!== null` can be reverted to as it was by ignoring `strict-boolean-expressions`. The previous check already did that in a cleaner way. We don't want to uglify our code for zero reason.
- These `String` coercions are also obnoxious, what rule is causing you to add them? According to [the `path.join` docs](https://nodejs.org/docs/latest/api/path.html#pathjoinpaths), that function returns a `string`, guaranteed, so why are we calling `String` on it? Whatever rule is making you do this, please see if you can ignore it too

Reply:
I agree with @taoeffect here, but like I said earlier, I don't think the issue is so much the rules but the configuration.